paas_app_charmer/_gunicorn/charm.py

```diff
@@ -81,6 +89,12 @@ def __init__(self, framework: ops.Framework, wsgi_framework: str) -> None:
         else:
             self._s3 = None
 
+        if "saml" in requires and requires["saml"].interface_name == "saml":
+            self._saml = SamlRequires(self)
+            self.framework.observe(self._saml.on.saml_data_available, self._on_saml_data_available)
+        else:
+            self._saml = None
+
```

Comment:
If the library was not installed (with `fetch-lib`), it will output a message in stderr with something like:
```
Missing charm library, please run `charmcraft fetch-lib charms.saml_integrator.v0.saml`
Traceback (most recent call last):
  File "/home/ubuntu/github/canonical/paas-app-charmer/paas_app_charmer/_gunicorn/charm.py", line 39, in <module>
    from charms.saml_integrator.v0.saml import SamlRequires
ModuleNotFoundError: No module named 'charms.saml_integrator.v0.saml'
```
and also, if the `saml` library is in `charmcraft.yaml`, then after that there will be an exception like: `NameError: name 'SamlRequires' is not defined`.


However, this will only happen if the charmcraft.yaml includes the `saml` library and the user did not install the saml library with `fetch-lib` (or in the future automatically with `fetch-libs`).

As the reason for this was to not introduce a backward incompatible change  (old users of paas-app-charmer failing because of redis library not installed because they did not execute fetch-libs), and it works if the saml library is included, I am not totally sure it is necessary to catch again this. WDYT?

Reply:
I was just thinking that those messages that are already there are generated on initial deployment. Let's say someone has this running for a while and then tries to use SAML, would a message be generated at that time? Thats what I was trying to think about. I guess this message is going to get printed on every execution hook actually

Reply:
Yeah, the error message (the one because of the import that instructs to install the lib) should be generated on every juju invocation, with level error, it will be quite visible if the user runs a `juju debug-log`